lib/api/controllers/document.js

```diff
@@ -466,6 +467,35 @@ class DocumentController extends NativeController {
     return { ids: documents.map(d => d._id) };
   }
 
+  /**
+   * Update several documents matching a query through the persistent layer
+   * 
+   * @param {Request} request
+   * @returns {Promise<Object} 
+   */
+  async updateByQuery (request) {
+    const
+      query = this.getBodyObject(request, 'query'),
+      changes = this.getBodyObject(request, 'changes'),
+      refresh = this.getString(request, 'refresh', 'false'),
+      source = this.getBoolean(request, 'source'),
+      { index, collection } = this.getIndexAndCollection(request);
+
+    const result = await this.publicStorage.updateByQuery(
+      index,
+      collection,
+      query,
+      changes,
+      { refresh });
+
+    await this.kuzzle.notifier.notifyDocumentMChanges(request, result.documents);
+
+    if (!source) {
+      result.documents.forEach(d => (d._source = null));
```

Comment:
A property set to `null` will be serialized by the JSON

```suggestion
      result.documents.forEach(d => (d._source = undefined));
```

Reply:
Sonarqube does not allow it

Reply:
Rule deactivated